src/ConductorSharp.Engine/Extensions/WorkflowEngineBuilder.cs

```diff
@@ -45,11 +49,20 @@ public class WorkflowEngineBuilder : IWorkflowEngineBuilder, IWorkflowEngineExec
 
             _builder.RegisterType<ExecutionManager>().SingleInstance();
 
+            _builder.RegisterType<ConductorSharpExecutionContext>().InstancePerLifetimeScope();
+
             _builder.RegisterGeneric(typeof(ValidationBehavior<,>)).As(typeof(IPipelineBehavior<,>));
 
             //_builder.RegisterGeneric(typeof(ErrorHandlingBehavior<, >)).As(typeof(IPipelineBehavior<, >));
 
             return this;
         }
+
+        public IWorkflowEngineExecutionManager AddRequestResponseLogging()
+        {
+            _builder.RegisterGeneric(typeof(RequestResponseLoggingBehavior<,>)).As(typeof(IPipelineBehavior<,>));
```

Comment:
This will result in validation executing after logging, it makes more sense to do this before validation is performed(validation could be complex and that could affect execution time which is not logged)

Reply:
Maybe also add `.AddValidation()` or something similar